src/OneSignal.ts

```diff
@@ -713,16 +714,16 @@ export default class OneSignal {
   /**
    * @PublicApi
    */
-  static async sendSelfNotification(title: string = 'OneSignal Test Message',
-                              message: string = 'This is an example notification.',
-                              url: string = `${new URL(location.href).origin}?_osp=do_not_open`,
-                              icon: URL,
-                              data: Map<String, any>,
-                              buttons: Array<NotificationActionButton>): Promise<void> {
+  static async sendSelfNotification(title = 'OneSignal Test Message',
+                              message = 'This is an example notification.',
+                              url = `${new URL(location.href).origin}?_osp=do_not_open`,
+                              icon?: string,
+                              data?: Record<string, any>,
+                              buttons?: Array<NotificationAction>): Promise<void> {
```

Comment:
Changing the type from `NotificationActionButton` to `NotificationAction` is a breaking change. The format is different between JS and the OneSignal API but we need to keep the backwards compatibility:
* https://documentation.onesignal.com/reference/push-channel-properties#action-buttons
* https://developer.mozilla.org/en-US/docs/Web/API/Notification/actions

Reply:
[That type](https://github.com/OneSignal/OneSignal-Website-SDK/blob/player-model-main/src/models/NotificationActionButton.ts#L3) has always been wrong then cause it doesn't match our API.

```ts
{
   action: string;
   title: string;
   icon: URL;
   url: URL;
}
```

This results in the error:

```
No overload matches this call.
  Overload 1 of 2, '(title: string, options?: NotificationOptions): Promise<void>', gave the following error.
    Argument of type '{ body: string; data: { data: Record<string, any>; url: string; buttons: NotificationActionButton[]; }; icon: string; actions: NotificationActionButton[]; }' is not assignable to parameter of type 'NotificationOptions'.
      Types of property 'actions' are incompatible.
        Type 'NotificationActionButton[]' is not assignable to type 'NotificationAction[]'.
          Type 'NotificationActionButton' is not assignable to type 'NotificationAction'.
            Types of property 'icon' are incompatible.
              Type 'URL' is not assignable to type 'string'.
  Overload 2 of 2, '(title: string, options?: NotificationOptions): Promise<void>', gave the following error.
    Argument of type '{ body: string; data: { data: Record<string, any>; url: string; buttons: NotificationActionButton[]; }; icon: string; actions: NotificationActionButton[]; }' is not assignable to parameter of type 'NotificationOptions'.

759       registration.showNotification(title, options);
```